databricks/sdk/config.py

```diff
@@ -425,13 +438,25 @@ def _known_file_config_loader(self):
             raise ValueError(f'resolve: {config_path} has no {profile} profile configured')
         raw_config = profiles[profile]
         logger.info(f'loading {profile} profile from {config_file}: {", ".join(raw_config.keys())}')
+
+        if self.compute is None and self.cluster_id is None:
+            if "compute" in raw_config and "cluster_id" in raw_config:
+                raise ValueError("Cannot specify both compute and cluster id in the config file")
         for k, v in raw_config.items():
             if k in self._inner:
                 # don't overwrite a value previously set
                 continue
+            if (k == "compute" and self.cluster_id is not None
+                    or k == "cluster_id" and self.compute is not None):
```

Comment:
Add some parenthesis so it's easier to read which operators are grouped.

```suggestion
            if (k == "compute" and self.cluster_id is not None)
                    or (k == "cluster_id" and self.compute is not None):
```